modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonAbstractCodegen.java

```diff
@@ -0,0 +1,454 @@
+import io.swagger.v3.oas.models.media.Schema;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.openapitools.codegen.*;
+import org.openapitools.codegen.utils.ModelUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.util.*;
+
+import static org.openapitools.codegen.utils.StringUtils.camelize;
+import static org.openapitools.codegen.utils.StringUtils.underscore;
+    
+public class PythonAbstractCodegen extends DefaultCodegen implements CodegenConfig {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PythonAbstractCodegen.class);
+
+    // nose is a python testing framework, we use pytest if USE_NOSE is unset
+    public static final String USE_NOSE = "useNose";
+
+    protected String packageName;
+    protected String packageVersion;
+    
+    protected Map<Character, String> regexModifiers;
+    protected boolean useNose = Boolean.FALSE;
+    
+    protected String pythonSrcRoot;
+
+    public PythonAbstractCodegen(String templateDirectory) {
+        super();
+
+        modelPackage = "models";
+
+        languageSpecificPrimitives.clear();
+        languageSpecificPrimitives.add("int");
+        languageSpecificPrimitives.add("float");
+        languageSpecificPrimitives.add("list");
+        languageSpecificPrimitives.add("dict");
+        languageSpecificPrimitives.add("bool");
+        languageSpecificPrimitives.add("str");
+        languageSpecificPrimitives.add("datetime");
+        languageSpecificPrimitives.add("date");
+        languageSpecificPrimitives.add("file");
+        languageSpecificPrimitives.add("object");
+        languageSpecificPrimitives.add("file");
+        languageSpecificPrimitives.add("byte");
+        languageSpecificPrimitives.add("bytearray");
+
+        typeMapping.clear();
+        typeMapping.put("integer", "int");
+        typeMapping.put("float", "float");
+        typeMapping.put("number", "float");
+        typeMapping.put("long", "int");
+        typeMapping.put("double", "float");
+        typeMapping.put("array", "List");
+        typeMapping.put("map", "Dict");
+        typeMapping.put("boolean", "bool");
+        typeMapping.put("string", "str");
+        typeMapping.put("date", "date");
+        typeMapping.put("DateTime", "datetime");
+        typeMapping.put("object", "object");
+        typeMapping.put("file", "file");
+        typeMapping.put("byte", "bytearray");
+        typeMapping.put("ByteArray", "bytearray");
+        // map uuid to string for the time being
+        typeMapping.put("UUID", "str");
+        typeMapping.put("URI", "str");
+
+
+        // clear import mapping (from default generator) as it does not contain any
+        // Python specific mappings.
+        importMapping.clear();
+        //
+        // Add the type mappings
+        importMapping.put("List", "from typing import List");
+        importMapping.put("Dict", "from typing import Dict");
```

Comment:
How does this work for python 2.7?
Don't we currently support both 2.7 and 3+? in the near future there is a plan to drop 2.7
What are your thoughts here @wing328 @jimschubert should 5.0.0 drop python 2.7 support?

The Python and python-experimental generators support python 2.7
How about defining these python 3 imports only in generators which exclusively produce python3 code?

Reply:
This works in python2.7 per our conversation

Reply:
This work in python python 2.7.18 and later only
@mwilby can you add an import that works for python 2.7 versions less than 18?

Reply:
This would be slightly problematic and a little bit strange. As far as I know the typing backport goes back at lest to 2016, so it should predate 2.7.18.

If not, in theory we would need a flag on the python version, but I don't think its necessary in the current python generator. In this case the types are represented as strings and the import is not included in the template, so the generated code will still be valid.

However, it is still an issue for the experimental version. I will look into it when I have a chance.

Reply:
When I tested it locally, typing does not exist in python 2.7.15 per:
```
Python 2.7.15 (default, Oct  2 2018, 11:47:18) 
[GCC 4.2.1 Compatible Apple LLVM 10.0.0 (clang-1000.11.45.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import typing
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named typing
```
Thank you!

Reply:
OK. I have cross checked and currently the python generator only represents the types as strings. The associated templates do not use the import strings. This means the code related to types will work in earlier version of 2.7, without the typing library. Of course this does not help with the experimental version.
I found documentation going as far back a 2.7.6 for using the typing library, but that just means it did exist at one time. The problem with unsupported systems, is that they decay exponentially and this may be one of those cases. The typing library was not shipped with the core system for 2.7 (I think it may be in 2.7.18, but I am not sure), it has to be pip installed and if the support libraries are not there any more, then its not going to work.
Being very pragmatic, I would be inclined to say support in 2.7.18 is sufficient. Any software that is still working on a earlier version is likely to fail because of other libraries, so it becomes a legitimate question of how much effort goes into supporting something that is likely to fail for other reasons.
I agree by saying this I am trying to weasel out of the problem, but it doesn't make it any the less true.